### DISCUSSION TITLE
ISFS and the future.

### DISCUSSION THREAD

**Discussion by Taulim** [intersystems-community/vscode-objectscript#573] [General]
Hello everyone.

I saw that `isfs` is being seems as a [*transition* feature](https://github.com/intersystems-community/vscode-objectscript/issues/568#issuecomment-775379059).  

While I like the idea of having all the code locally (that makes it easier to source control) it is perfect for developing while alone I know this is not going to work in the long run on how cache/iris work.  
One of the big things about Studio is that opening the super-server port you have access to everything on that namespace without having it on your machine (great for support and small emergency fixes or shared servers - I dislike this one...).  

If we only look to `isfs` as transitional we will end up losing this *feature* and Studio will always be needed.  

I'm using `isfs` exclusively for some time now (I even hide the cache icon from my vscode) and it has been working very well.  
With some exceptions (global search, multiple namespaces, some rare .csp compile errors) I mostly keep my Studio closed, the same studio hooks that we had before still works now so no change was required server-side (and can go back to studio anytime if anything goes wrong, like my containers not launching language-server xD ).  

Even if the `isfs` need to be split in another extension I hope we can continue to have it as an alternative and not a transitional feature (unless IS is planning to discontinue the Atelier API without anything to its place, if so I think it is a mistake).  

Thanks for your time.

**Comment by gjsjohnmurray** [comment]
@Taulim the comment by @daimor that you linked to states it is a personal opinion. I take the opposite view, and agree with you about `isfs` being valuable in the long term too.

Pinging @timleavitt who may be interested in this discussion.

**Comment by isc-rsingh** [comment]
@Taulim as you can see there are differences of opinion on the idea that this is a transitional feature. For some development workflows, this will be the case as teams adopt a client-side source control approach. But as long as the ISFS server-side approach is a common workflow with InterSystems customers, we will continue to support it. In fact this feature is getting enhancements on a regular basis!
